src/daemon/daemon.cpp

```diff
@@ -135,7 +135,7 @@ bool t_daemon::run(bool interactive)
 
     mp_internals->p2p.run(); // blocks until p2p goes down
 
-    if (interactive)
+    if (rpc_commands)
```

Comment:
Interesting. Would this have resulted in any errors or odd behaviour that we just haven't spotted before now?